File: Codechef/Oct/SAKTAN.py
from collections import defaultdict

def counter(dic):
	fOdd = 0
	fEven = 0;
	for f, s in mpX.items:
	    if s & 1:
	        fOdd += 1
	    else:
	        fEven += 1
	return [fOdd, fEven]

t = int(raw_input())
for i in range(0, t):
	n, m, q = list(map(int, raw_input().split()))
	mpX = defaultdict(bool)
	mpY = defaultdict(bool)
	for k in range(0, q):
		x, y = list(map(int, raw_input().split()))
		mpX[x] = not mpX[x]
		mpY[y] = not mpY[y]

	fO, fE = counter(mpX)
	sO, sE = counter(mpY)

	res = fO * sE + sO * fE + fO * (m - len(mpY)) + sO * (n - len(mpX))
	print res

File: Codechef/Oct/SAKTAN.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <algorithm>
#include <cmath>
#include <queue>
#include <cstring>
#include <sstream>
#include <map>
#include <climits>
#include <set>

using namespace std;


int main(){
// 	freopen("input.txt", "r", stdin);

	int t;
	cin>>t;
	while (t--){
		int n, m, q, x, y;
		cin>>n>>m>>q;
		map<int,bool> mpX;
		map<int,bool> mpY;
		for (int i=0; i<q; i++){
			cin>>x>>y;
			mpX[x] = !mpX[x];
			mpY[y] = !mpY[y];
		}
		int fOdd = 0, fEven = 0;
		for (auto i : mpX){
		    if (i.second & 1){
		        fOdd++;
		    }else{
		        fEven++;
		    }
		}
		
		int sOdd = 0, sEven = 0;
		for (auto i : mpY){
		    if (i.second & 1){
		        sOdd++;
		    }else{
		        sEven++;
		    }
		}
		int res = fOdd * sEven + fEven * sOdd + fOdd * (m - mpY.size()) + sOdd * (n - mpX.size());
		cout<<res<<endl;
	}
 return 0;
}